### DISCUSSION TITLE
Identification of actual container format from data stream

### DISCUSSION THREAD

**Discussion by jubilant-enigma** [SixLabors/ImageSharp#1867] [Q&A] [answered]
Unless I am mistaken, `Image.Identify` returns a `IImageInfo` which contains the bit depth, resolution, and metadata of an image. It does not seem to contain the actual type of the filetype/container of the file that was queried. 

For example, suppose I want to use ImageSharp to detect and rename `.jpg` files that have been incorrectly renamed to `.png`. The actual detection of this is relatively simple, it just involves looking at the first few bytes of the file. However, I cannot see see a way to actually detect this from the `Identify` API. Should the `IImageInfo` interface include a member variable that specifies the container format?

**Comment by tocsoft** [comment, marked as answer]
You want to use the overload that returns an `IImageFormat` as an `out` parameter.

i.e. call [`IImageInfo Identify(Stream stream, out IImageFormat format)`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Image.html#SixLabors_ImageSharp_Image_Identify_System_IO_Stream_SixLabors_ImageSharp_Formats_IImageFormat__)


[`IImageFormat`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Formats.IImageFormat.html) then gives you access to things like the default mime type or file extensions all based on the actual format and not on the file path with the wrong extension.
